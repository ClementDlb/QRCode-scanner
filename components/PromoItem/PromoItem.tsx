import React from 'react'
import { View, Text, TouchableOpacity, Image, StyleSheet, Alert } from 'react-native'

const seeImg = require("../../assets/images/see.png")

function PromoItem(props) {
    return (
      <View style={styles.item}>
        <Text style={styles.itemText} testID={'Nomtext'}>{props.nom}</Text>
        <Text style={styles.amount} testID={'Montanttext'}>{props.montant}%</Text>
        <Text style={styles.expDate} testID={'ExpireAttext'}>{props.expireAt}</Text>
        <TouchableOpacity style={styles.seeButton} testID={'AlertCodeButton'} 
          onPress={()=>Alert.alert(props.code)}><Image style={styles.seeImg} source={seeImg}></Image>
        </TouchableOpacity>
        <Image style={styles.couponImg} source={require('../../assets/images/coupon.png')}></Image>
      </View>
    );
  }

  export default PromoItem;

  const styles = StyleSheet.create({
    item: {
      flexDirection: 'row', alignItems: 'flex-start', paddingHorizontal: 0, paddingTop: 100, alignContent:'center'
    },
    itemText: {
      fontSize: 28, position:'absolute', zIndex:1, fontFamily:'Cartoon', color:"#FFF", paddingLeft:60, paddingTop:10
    },
    amount:{
      padding:5, fontSize: 24, position:'absolute', zIndex:2, fontFamily:'Chill', alignSelf:'center', paddingLeft:75,paddingTop:15, color:"#FFF"
    },
    expDate:{
      padding:5, fontSize: 14, position:'absolute', zIndex:2, fontFamily:'space-mono', alignSelf:'center', paddingLeft:70, paddingTop:55
    },
    seeButton:{
      position:'absolute', zIndex:3, width:50, left:250, marginTop:52
    },
    couponImg:{
      position:'absolute',
      width:380,
      height:90,
    },
    seeImg:{
      position:'absolute', width:40,height:23, alignItems:'center'
    }
  })