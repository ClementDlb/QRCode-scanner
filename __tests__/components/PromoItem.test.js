import React from 'react';
import { Alert } from 'react-native';
import PromoItem from '../../components/PromoItem/PromoItem'
import { render, cleanup, fireEvent } from '@testing-library/react-native';

afterEach(cleanup);

describe('<PromoItem />', () => {
    it('snapshotPromoItem', () => {
        const renderered = render(<PromoItem nom="test" montant={50} expireAt='24/01/2021' code="HIVER33"></PromoItem>).toJSON();
        expect(renderered).toMatchSnapshot();
    });

    it('doit rendre le nom correctement', () => {
        const rendered = render(<PromoItem nom="test"></PromoItem>);
        const textComponent = rendered.getByTestId('Nomtext')

        expect(textComponent.props.children).toEqual('test');
    });

    it('doit rendre le montant correctement', () => {
        const rendered = render(<PromoItem montant={50}></PromoItem>);
        const textComponent = rendered.getByTestId('Montanttext')

        expect(textComponent.props.children).toEqual([50, "%"]);
    });

    it("doit rendre la date d'expiration correctement", () => {
        const rendered = render(<PromoItem expireAt="24/01/2021"></PromoItem>);
        const textComponent = rendered.getByTestId('ExpireAttext')

        expect(textComponent.props.children).toEqual('24/01/2021');
    });

    it("doit alert le code onPress", () => {
        jest.spyOn(Alert, 'alert')
        const rendered = render(<PromoItem code="HIVER33"></PromoItem>);
        const buttonComponent = rendered.getByTestId('AlertCodeButton')
        fireEvent(buttonComponent, 'press');

        expect(Alert.alert).toHaveBeenCalledWith("HIVER33")
    });

})