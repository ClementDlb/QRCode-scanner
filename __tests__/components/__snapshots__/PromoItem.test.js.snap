// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PromoItem /> snapshotPromoItem 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "alignItems": "flex-start",
      "flexDirection": "row",
      "paddingHorizontal": 0,
      "paddingTop": 100,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontFamily": "Cartoon",
        "fontSize": 28,
        "paddingLeft": 60,
        "paddingTop": 10,
        "position": "absolute",
        "zIndex": 1,
      }
    }
    testID="Nomtext"
  >
    test
  </Text>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#FFF",
        "fontFamily": "Chill",
        "fontSize": 24,
        "padding": 5,
        "paddingLeft": 75,
        "paddingTop": 15,
        "position": "absolute",
        "zIndex": 2,
      }
    }
    testID="Montanttext"
  >
    50
    %
  </Text>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontFamily": "space-mono",
        "fontSize": 14,
        "padding": 5,
        "paddingLeft": 70,
        "paddingTop": 55,
        "position": "absolute",
        "zIndex": 2,
      }
    }
    testID="ExpireAttext"
  >
    24/01/2021
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 250,
        "marginTop": 52,
        "opacity": 1,
        "position": "absolute",
        "width": 50,
        "zIndex": 3,
      }
    }
    testID="AlertCodeButton"
  >
    <Image
      source={1}
      style={
        Object {
          "alignItems": "center",
          "height": 23,
          "position": "absolute",
          "width": 40,
        }
      }
    />
  </View>
  <Image
    source={1}
    style={
      Object {
        "height": 90,
        "position": "absolute",
        "width": 380,
      }
    }
  />
</View>
`;
